certbot/plugins/standalone_test.py

```diff
@@ -106,13 +106,22 @@ def test_dvsni(self):
         self.assertEqual("tls-sni-01,http-01", self._call("dvsni,http-01"))
 
 
+def get_open_port():
+    """Gets an open port number from the OS."""
+    open_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)
```

Comment:
I may or may not gotten my understanding of what's going on here right, or explained it right; we should probably sync up our understanding of all of the cases in person. And from a high level this is all pretty corner case-y and not worth doing extra work for. So merging as is :)